### ISSUE TITLE
EMSL study identifiers 

### ISSUE THREAD

**Issue by aclum** [microbiomedata/nmdc-schema#927]
Pending discussions with EMSL create an nmdc schema slot for EMSL study identifiers (ie https://www.emsl.pnnl.gov/project/60141)

There is a curie registered at bioregistry
https://bioregistry.io/emsl.project

Related to https://github.com/microbiomedata/nmdc-schema/issues/868

**Comment by mslarae13** [comment]
From @lamccue 
Check with EMSL leadership (George and Ken) about the bioregistry curie for EMSL. The emsl.project prefix that was registered is no problem from their viewpoint.  We agreed that if NMDC identifies a desire/need to have more prefixes (for samples or other things), then the process for us will be to talk to George and Ken about it first - what the prefix is and how it will be used, etc. They can decide if it is something that they also want to incorporate into the NEXUS schema, and then we decide how to proceed and who does the registering.

**Comment by mslarae13** [comment]
With that, we'd like to get EMSL projects linked in the data portal on the study landing pages at the "External Resources" -> "Additional data".

@aclum or @turbomam  or @naglepuff or @dwinston Can someone tell what the class-slot is I need to put these links under? I attempted to use the API but was unsuccessful in finding the slot.

I assume this will need to be added to mongo via change sheet?

**Comment by mslarae13** [comment]
@naglepuff when we do get these added, they should appear here
![Screen Shot 2023-06-01 at 7 32 13 PM](https://github.com/microbiomedata/nmdc-schema/assets/44581429/9152d79b-069d-48b7-a38b-0175306684db)

With this logo
![EML_Logo-Light](https://github.com/microbiomedata/nmdc-schema/assets/44581429/6996950b-1cd6-41cd-8946-99e035330dde)

Unless someone has another opinion? (I will check with Lee Ann)

**Comment by mslarae13** [comment]
@turbomam this emsl.identifiers curie should probably have it's own slot like GOLD, 

https://microbiomedata.github.io/nmdc-schema/study_identifiers/

I believe this is where these links comes from?

**Comment by aclum** [comment]
That slot hasn't been used yet. That could be used, if we do at some point it should be updated so it inherits from alternative_identifiers and be enforced as a curie now that we have one. @turbomam  let us know if you want to do that now under this ticket, that seems cleaner since we haven't used the slot yet or if this ticket should be closed and we just need a nmdc-server repo ticket to get this information displayed.

**Comment by mslarae13** [comment]
https://github.com/microbiomedata/nmdc-schema/issues/960

@turbomam  is adding emsl_project_identifiers to study_identifiers: https://microbiomedata.github.io/nmdc-schema/study_identifiers/ 

However, EMSL uses study, proposal, and project rather interchangeably. 

Addition of this new slot, will include a todo, to add alias.

- [ ] What are the aliases? 
> Montana to check with NEXUS
- [ ] Add aliases

**Comment by turbomam** [comment]
Here are the authority-based mixins in src/schema/external_identifiers.yaml, and the object-type-based identifier slots that they are mixed into.

- _Some of this hierarchy is lost in the submission-schema_
-  `jgi_portal_identifiers` and `neon_identifiers`haven't haven't been mixed into any usable identifiers yet. They're just abstract now.
- there isn't any `emsl_identifiers` mixin yet! But there are some slots with 'emsl_' in their names, defined in schema files other than src/schema/external_identifiers.yaml

----

- src/schema/portal/emsl.yaml:
  - `emsl_store_temp`
- src/schema/nmdc.yaml:
  - `emsl_biosample_identifiers` (move to src/schema/external_identifiers.yaml?)
  - `emsl_project_identifier` (move to src/schema/external_identifiers.yaml?)
  - `emsl_proposal_doi`
  - `emsl_proposal_identifier` (move to src/schema/external_identifiers.yaml?)
 
----

- `gnps_identifiers` is only mixed into `study_identifiers` to get `gnps_task_identifiers`
  - [alternative_identifiers](https://microbiomedata.github.io/nmdc-schema/alternative_identifiers/)
    - [external_database_identifiers](https://microbiomedata.github.io/nmdc-schema/external_database_identifiers/)
      - [study_identifiers](https://microbiomedata.github.io/nmdc-schema/study_identifiers/)
        - gnps_task_identifiers [ [gnps_identifiers](https://microbiomedata.github.io/nmdc-schema/gnps_identifiers/)]
- `gold_identifiers`
  - mixed into many identifiers. see below.
- `insdc_identifiers`
  - mixed into many identifiers. see below.
- `jgi_portal_identifiers`
  - **not mixed into anything yet**
- `massive_identifiers` is only mixed into `study_identifiers` to get `massive_study_identifiers`
  - [alternative_identifiers](https://microbiomedata.github.io/nmdc-schema/alternative_identifiers/)
    - [external_database_identifiers](https://microbiomedata.github.io/nmdc-schema/external_database_identifiers/)
      - [study_identifiers](https://microbiomedata.github.io/nmdc-schema/study_identifiers/)
        - massive_study_identifiers [ [massive_identifiers](https://microbiomedata.github.io/nmdc-schema/massive_identifiers/)]
- `mgnify_identifiers` is only mixed into `analysis_identifiers` to get `mgnify_analysis_identifiers`
  - [alternative_identifiers](https://microbiomedata.github.io/nmdc-schema/alternative_identifiers/)
    - [external_database_identifiers](https://microbiomedata.github.io/nmdc-schema/external_database_identifiers/)
      - [analysis_identifiers](https://microbiomedata.github.io/nmdc-schema/analysis_identifiers/)
        - mgnify_analysis_identifiers [ [mgnify_identifiers](https://microbiomedata.github.io/nmdc-schema/mgnify_identifiers/)]
- `neon_identifiers`
  - **not mixed into anything yet**

## gold_identifiers
mixed into | description | range | domain
-- | -- | -- | --
gold_study_identifiers | identifiers for corresponding project(s) in GOLD | None | Study
gold_biosample_identifiers | identifiers for corresponding sample in GOLD | None | Biosample
gold_sequencing_project_identifiers | identifiers for corresponding sequencing project in GOLD | None | OmicsProcessing
gold_analysis_project_identifiers | identifiers for corresponding analysis project in GOLD | None | MetagenomeAnnotationActivity, MetatranscriptomeAnnotationActivity


## insdc_identifiers
mixed into | description | range | domain
-- | -- | -- | --
insdc_sra_ena_study_identifiers | identifiers for corresponding project in INSDC SRA / ENA | None |  
insdc_bioproject_identifiers | identifiers for corresponding project in INSDC Bioproject | None | Study
insdc_biosample_identifiers | identifiers for corresponding sample in INSDC | None | Biosample
insdc_secondary_sample_identifiers | secondary identifiers for corresponding sample in INSDC | None |  
insdc_experiment_identifiers |   | None | OmicsProcessing
insdc_analysis_identifiers |   | None |  
insdc_assembly_identifiers |   | None | MetagenomeAssembly, MetatranscriptomeAssembly

**Comment by turbomam** [comment]
Here are the authority-based mixins in src/schema/external_identifiers.yaml, and the object-type-based identifier slots that they are mixed into.

- _Some of this hierarchy is lost in the submission-schema_
-  `jgi_portal_identifiers` and `neon_identifiers`haven't haven't been mixed into any usable identifiers yet. They're just abstract now.
- there isn't any `emsl_identifiers` mixin yet! But there are some slots with 'emsl_' in their names, defined in schema files other than src/schema/external_identifiers.yaml

----

- src/schema/portal/emsl.yaml:
  - `emsl_store_temp`
- src/schema/nmdc.yaml:
  - `emsl_biosample_identifiers` (move to src/schema/external_identifiers.yaml?)
  - `emsl_project_identifier` (move to src/schema/external_identifiers.yaml?)
  - `emsl_proposal_doi`
  - `emsl_proposal_identifier` (move to src/schema/external_identifiers.yaml?)
 
----

- `gnps_identifiers` is only mixed into `study_identifiers` to get `gnps_task_identifiers`
  - [alternative_identifiers](https://microbiomedata.github.io/nmdc-schema/alternative_identifiers/)
    - [external_database_identifiers](https://microbiomedata.github.io/nmdc-schema/external_database_identifiers/)
      - [study_identifiers](https://microbiomedata.github.io/nmdc-schema/study_identifiers/)
        - gnps_task_identifiers [ [gnps_identifiers](https://microbiomedata.github.io/nmdc-schema/gnps_identifiers/)]
- `gold_identifiers`
  - mixed into many identifiers. see below.
- `insdc_identifiers`
  - mixed into many identifiers. see below.
- `jgi_portal_identifiers`
  - **not mixed into anything yet**
- `massive_identifiers` is only mixed into `study_identifiers` to get `massive_study_identifiers`
  - [alternative_identifiers](https://microbiomedata.github.io/nmdc-schema/alternative_identifiers/)
    - [external_database_identifiers](https://microbiomedata.github.io/nmdc-schema/external_database_identifiers/)
      - [study_identifiers](https://microbiomedata.github.io/nmdc-schema/study_identifiers/)
        - massive_study_identifiers [ [massive_identifiers](https://microbiomedata.github.io/nmdc-schema/massive_identifiers/)]
- `mgnify_identifiers` is only mixed into `analysis_identifiers` to get `mgnify_analysis_identifiers`
  - [alternative_identifiers](https://microbiomedata.github.io/nmdc-schema/alternative_identifiers/)
    - [external_database_identifiers](https://microbiomedata.github.io/nmdc-schema/external_database_identifiers/)
      - [analysis_identifiers](https://microbiomedata.github.io/nmdc-schema/analysis_identifiers/)
        - mgnify_analysis_identifiers [ [mgnify_identifiers](https://microbiomedata.github.io/nmdc-schema/mgnify_identifiers/)]
- `neon_identifiers`
  - **not mixed into anything yet**

## gold_identifiers
mixed into | description | range | domain
-- | -- | -- | --
gold_study_identifiers | identifiers for corresponding project(s) in GOLD | None | Study
gold_biosample_identifiers | identifiers for corresponding sample in GOLD | None | Biosample
gold_sequencing_project_identifiers | identifiers for corresponding sequencing project in GOLD | None | OmicsProcessing
gold_analysis_project_identifiers | identifiers for corresponding analysis project in GOLD | None | MetagenomeAnnotationActivity, MetatranscriptomeAnnotationActivity


## insdc_identifiers
mixed into | description | range | domain
-- | -- | -- | --
insdc_sra_ena_study_identifiers | identifiers for corresponding project in INSDC SRA / ENA | None |  
insdc_bioproject_identifiers | identifiers for corresponding project in INSDC Bioproject | None | Study
insdc_biosample_identifiers | identifiers for corresponding sample in INSDC | None | Biosample
insdc_secondary_sample_identifiers | secondary identifiers for corresponding sample in INSDC | None |  
insdc_experiment_identifiers |   | None | OmicsProcessing
insdc_analysis_identifiers |   | None |  
insdc_assembly_identifiers |   | None | MetagenomeAssembly, MetatranscriptomeAssembly

**Comment by turbomam** [comment]
----

- src/schema/portal/emsl.yaml:
  - `emsl_store_temp`
- src/schema/nmdc.yaml:
  - `emsl_biosample_identifiers` (move to src/schema/external_identifiers.yaml?)
  - `emsl_project_identifier` (move to src/schema/external_identifiers.yaml?)
  - `emsl_proposal_doi`
  - `emsl_proposal_identifier` (move to src/schema/external_identifiers.yaml?)

**Comment by turbomam** [comment]
@mslarae13 @SamuelPurvine 

Here's the ting I'm most confused about:

Class `Study` uses both

- emsl_project_identifier
- emsl_proposal_identifier

If a `Study` needs to mention multiple EMSL entities, these slots need better differentiating criteria.

**Comment by turbomam** [comment]
We should also rename `emsl_proposal_identifier` to `emsl_proposal_identifiers` because all alternative identifiers take a list of values

**Comment by SamuelPurvine** [comment]
let me preface my comment by saying I wish I had the power to "make it so". Meaning I'm not certain I am a/the decider. Lee Ann showed preference for keeping emsl_project_identifier as the slot, and setting emsl_proposal_identifier as an alias if that's possible. The description would change to something like, "The project number assigned to the EMSL awarded study proposal  that relates to that which is represented in NMDC" and drop any description for emsl_proposal_identifier. As to renaming it to the plural sense, that's dealer's choice :) 

Part of the confusion is there is not yet consensus on EMSL's part as to which verbiage is the correct one to use. Proposal is what we've used for years, as a written proposal is the thing that is actually sent by an EMSL User so may as well just call it that. However, PNNL culture speaks in terms of projects when discussing a body of work, irrespective of whether it was derived from a proposal we sent out, or was sent to us, or was broken out internally with no particular proposal associated. I dimly recall semi-pitched battle when I started here about whether we should/could refer to EMSL work as a proposal or a project, with the decision made to call these things "proposal" to reduce confusion with the higher level entities known as projects (in PNNL parlance, the entire EMSL endeavor is a project). Sorry for the pedantry, it's increasingly becoming part of by nature!

**Comment by mslarae13** [comment]
@turbomam 

IMO, 
> emsl_proposal_doi
> emsl_proposal_identifier (move to src/schema/external_identifiers.yaml?)

Can be depricated and we should just have emsl_project_identifier where we link the DOIs and have alias for proposal and project_doi and award_doi

emsl_project_identifer needs to be under alternative identifiers > external database identifiers
https://microbiomedata.github.io/nmdc-schema/study_identifiers/ 

With the DOI, you can get the EMSL project (proposal) # and that will also provide the information you'd need to find it on DMS in EMSL (@SamuelPurvine )

**Comment by turbomam** [comment]
Thanks. Are you saying that you only want `emsl_project_identifier`,  which would be a sub-property of `study_identifiers`?


TL;DR: I don't see how we can consider `emsl_project_identifier` a `study_identiifers` and use it to link to a DOI

The range of `study_identiifers` is uri or curie, but I have applied a validation pattern that required the curie form.

Is this bit from you post above supposed to be a valid value for `emsl_project_identifier`?

> emsl.project: https://doi.org/10.46936/intm.proj.2021.60141/60000423

I would expect the value to something like 'emsl.project:60141', where `emsl.project` resolves to 'https://bioregistry.io/reference/emsl.project:'

So the expanded URL would be https://bioregistry.io/reference/emsl.project:60141, which redirects to https://www.emsl.pnnl.gov/project/60141,

**Comment by aclum** [comment]
@turbomam Ive made example data https://github.com/microbiomedata/nmdc-schema/blob/issue-927-emsl-doi/src/data/valid/Study-emsl-doi.yaml

currently modeled as a single value but pending input @mslarae13 we may want this to be a list

**Comment by aclum** [comment]
@mslarae13  confirmed that slots should allow a list and requested a change to the slot name. I've updated the [example valid data ](https://github.com/microbiomedata/nmdc-schema/blob/issue-927-emsl-doi/src/data/valid/Study-emsl-doi.yaml and )
created some [example invalid data ](https://github.com/microbiomedata/nmdc-schema/blob/issue-927-emsl-doi/src/data/invalid/Study-bad-emsl-doi-slot-name.yaml)that uses the old slot name 

to summarize the slot should have the following:
- name emsl_project_doi
- use a prefix of doi
- use identifers.org or bioregistry.io as where the prefix is registered
- accept a list of dois

@turbomam

**Comment by turbomam** [comment]
Your example 

- uses a new `emsl_project_doi` slot to associate study 'nmdc:sty-11-ab' with a resource with the curie doi:10.46936/intm.proj.2021.60141/60000423
- the schema will expand prefix 'doi' to a base url of 'https://bioregistry.io/doi:', so 'doi:10.46936/intm.proj.2021.60141/60000423' will expand to 'https://bioregistry.io/doi:10.46936/intm.proj.2021.60141/60000423'
- bioregistry and/or OSTI expand that to https://www.osti.gov/award-doi-service/biblio/10.46936/intm.proj.2021.60141/60000423

the following will be removed from the schema
- emsl_project_identifier
- emsl_proposal_doi
- emsl_proposal_identifier

**Comment by turbomam** [comment]
Remaining question: should `emsl_project_doi ` be placed in the [alternative_identifiers](https://microbiomedata.github.io/nmdc-schema/alternative_identifiers) hierarchy? Perhaps in [study_identifiers](https://microbiomedata.github.io/nmdc-schema/study_identifiers/) ? Maybe even taking the place that will be vacated by [emsl_proposal_identifier](https://microbiomedata.github.io/nmdc-schema/emsl_proposal_identifier/) ?

**Comment by aclum** [comment]
blocked on data portal study page revamp squad.

**Comment by mslarae13** [comment]
emsl_project_doi & emsl_proposal_identifier were removed. 

EMSL study curie : https://bioregistry.io/registry/emsl.project 

Need to add EMSL study ID numbers to the. Add (un comment out) emsl_project_identifer to the schema. Study alternative_identifiers with a range of the EMSL curie.

**Comment by mslarae13** [comment]
Prefix expansion www.emsl.pnnl.gov/project/ 

So when the 5 digit project number is associated with the prefix (emsl.project:60141) it resolves to https://www.emsl.pnnl.gov/project/60141 

full curie = emsl.project:60141 

and the registered prefix is at bioregistry -- https://bioregistry.io/registry/emsl.project

**Comment by turbomam** [comment]
@turbomam ^

**Comment by turbomam** [comment]
I'm going to summarize the situation in my own words, and say what I think I should do next.
- [the **prefix** `emsl.project` is associated with the **expansion** `https://bioregistry.io/emsl.project:` in nmdc.yaml](https://github.com/microbiomedata/nmdc-schema/blob/918056b61508b2094433abb15fbb0e1578fd014f/src/schema/nmdc.yaml#L74C3-L74)
- [`emsl.project` is **registered** as a prefix at https://bioregistry.io/](https://bioregistry.io/registry/emsl.project)
- `60141` is a valid EMSL Project ID. Abstractly, we would call that a valid local identifier.
- A **CURIe** (or compact URI) is a prefix + a local identifier, so at least NMDC and Bioregistry are saying that `emsl.project:60141` is a reasonable CURIe, which should be expanded to https://bioregistry.io/emsl.project:60141
- Bioregistry will **redirect** URLs like https://bioregistry.io/emsl.project:60141 to https://www.emsl.pnnl.gov/project/60141
  - this redirection increases complexity and overhead, and it requires coordination between NMDC, Bioregistry and ideally authorized people from EMSL.
  - the benefit is FAIRness, especially Findability, Interoperability, and Reuse. Hopefully other people who need to abbreviate EMSL Project IDs and page URLs will follow our example and use the `emsl.project` prefix and Bioregistry's redirection
